src/flow/domain/mod.rs

```diff
@@ -612,33 +637,44 @@ impl Domain {
                     .expect("migration replay path started with non-materialized node")
                     .lookup(&cols[..], &KeyType::Single(&key[0]));
 
-                let m = if let LookupResult::Some(rs) = rs {
+                if let LookupResult::Some(rs) = rs {
                     use std::iter::FromIterator;
-                    Some(Packet::ReplayPiece {
+                    let m = Some(Packet::ReplayPiece {
                              link: Link::new(source, path[0].0),
                              tag: tag,
-                             context: ReplayPieceContext::Partial { for_key: Vec::from(key) },
+                             context: ReplayPieceContext::Partial { for_key: Vec::from(key), ignore: false },
                              data: Records::from_iter(rs.into_iter().cloned()),
-                         })
+                             transaction_state: transaction_state,
+                    });
+                    (m, source, false)
+                } else if transaction_state.is_some() {
+                    let m = Some(Packet::ReplayPiece {
```

Comment:
I'd like a brief comment here saying "we need to forward a `ReplayPiece` for the timestamp we claimed"